xarray/backends/scipy_.py

```diff
@@ -61,8 +63,18 @@ def _open_scipy_netcdf(filename, mode, mmap, version):
         # it's a NetCDF3 bytestring
         filename = BytesIO(filename)
 
-    return scipy.io.netcdf_file(filename, mode=mode, mmap=mmap,
-                                version=version)
+    try:
+        ds = scipy.io.netcdf_file(filename, mode=mode, mmap=mmap,
+                                  version=version)
+    except TypeError as e:
+        # TODO: gzipped loading only works with NetCDF3 files.
+        if 'is not a valid NetCDF 3 file' in e.message:
```

Comment:
This should only be triggered when reading a gzipped file. Right now, it can be triggered whenever an invalid netCDF3 file is read, gzipped or not.

This should be easier to fix when you move `gzip.open` into this helper function (see my comment below).